### PR TITLE
 Geofencing feature file for Test Definition

### PR DESCRIPTION
**Pull request by mdomale**
#### What type of PR is this?

Add one of the following kinds:
* bug
* correction
* enhancement/feature - Test Definition for Geofencing Feature file
* cleanup
* documentation
* subproject management
* tests

#### Which issue(s) this PR fixes:

Fixes #

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

**Comment by mdomale** [comment]
@akoshunyadi  @shilpa-padgaonkar  @hdamker FYI

**Comment by mdomale** [comment]
duplicate hence closing

File: code/Test_definitions/Geofencing.feature [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+ @Geofencing 
+Feature: Operations to manage event subscription on geofencing events for leaving and entering in an area
+
+  @geofencing_01_Create_geofencing_subscription_for_a_device
+  Scenario:  Create geofencing subscription
+    Given they use the geofencing url
+    When they create subscription
+    Then Response code is 201
+
+  @geofencing_02_Operation_to_retrieve_list_of_subscriptions.--Not yet implemented
```
Comment (alpaycetin74) [inline code/Test_definitions/Geofencing.feature:33]: I guess a space is needed before the double dash, at the moment the string "--Not" is perceived as part of the expression starting with '@'

File: code/Test_definitions/Geofencing.feature [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    When they delete subscription for subscription-id
+    Then Response code is 204
+
+  @geofencing_05_Create_invalid_geofencing_subscription_for_a_device
+  Scenario:  Create geofencing subscription with invalid parameter
+    Given they use the geofencing url
+    When they create subscription with invalid parameter
+    Then Response code is 400
+
+  @geofencing_06_Get_invalid_geofencing_subscription_for_a_device
```
Comment (alpaycetin74) [inline code/Test_definitions/Geofencing.feature:58]: We need to clarify what "invalid" subscription id means.
If the id has a valid format, but unknown to the system, the proper response is 404.
If the id does not have a valid format, the proper response is 400.
Maybe we should have one test for each case.

File: code/Test_definitions/Geofencing.feature [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  Scenario:  Get geofencing subscription with invalid parameter
+    Given they use the geofencing url
+    When they get subscription with invalid subscription-id
+    Then Response code is 404
+
+
+  @geofencing_07_Delete_invalid_geofencing_subscription_for_a_device
```
Comment (alpaycetin74) [inline code/Test_definitions/Geofencing.feature:65]: Same as in GET case:
We need to clarify what "invalid" subscription id means.
If the id has a valid format, but unknown to the system, the proper response is 404.
If the id does not have a valid format, the proper response is 400.
Maybe we should have one test for each case.

**Pull request closed without merging**